generator/template.go

```diff
@@ -468,15 +468,24 @@ func renderURL(r *registry.Registry) func(method data.Method) string {
 	fieldNameFn := fieldName(r)
 	return func(method data.Method) string {
 		methodURL := method.URL
-		reg := regexp.MustCompile("{([^}]+)}")
+		// capture fields like {abc} or {abc=def/ghi/*}. Discard the pattern after the equal sign.
+		reg := regexp.MustCompile("{([^=}]+)=?([^}]+)?}")
 		matches := reg.FindAllStringSubmatch(methodURL, -1)
 		fieldsInPath := make([]string, 0, len(matches))
 		if len(matches) > 0 {
 			log.Debugf("url matches %v", matches)
 			for _, m := range matches {
 				expToReplace := m[0]
-				fieldName := fieldNameFn(m[1])
-				part := fmt.Sprintf(`${req["%s"]}`, fieldName)
+				// convert foo_bar.baz_qux to fieldName `fooBar.bazQux`, part `${req["fooBar"]["bazQux"]}`
+				subFields := strings.Split(m[1], ".")
+				var subFieldNames, partNames []string
+				for _, subField := range subFields {
+					subFieldName := fieldNameFn(subField)
+					subFieldNames = append(subFieldNames, subFieldName)
+					partNames = append(partNames, fmt.Sprintf(`["%s"]`, subFieldName))
```

Comment:
Because the request type generated for a nested requests makes the [fields optional](https://github.com/mbarrien/protoc-gen-grpc-gateway-ts/blob/07fb69db5d49d4f86b9b4b8577652ba3b0a1aa28/generator/template.go#L46-L51), we'll need to add optional chaining (`?.`) to this subField name append to get something like 
```ts
${req["fooBar"]?.["bazQux"]}
```

Reply:
Made the change to use the optional chaining on every nested field by changing the Join(partNames, "") to Join(partNames, "?.") a few lines below.